clients/fides-js/__tests__/lib/tcf/fidesString.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  decodeFidesString,
+  idsFromAcString,
+} from "../../../src/lib/tcf/fidesString";
+
+describe("decodeFidesString", () => {
+  it.each([
+    // Empty string
+    {
+      fidesString: "",
+      expected: { tc: "", ac: "" },
+    },
+    // TC string only
+    {
+      fidesString: "CPzvOIAPzvOIAGXABBENAUEAAACAAAAAAAAAAAAAAAAA.IAAA",
+      expected: {
+        tc: "CPzvOIAPzvOIAGXABBENAUEAAACAAAAAAAAAAAAAAAAA.IAAA",
+        ac: "",
+      },
+    },
+    // Without vendors disclosed
+    {
+      fidesString: "CPzvOIAPzvOIAGXABBENAUEAAACAAAAAAAAAAAAAAAAA",
+      expected: { tc: "CPzvOIAPzvOIAGXABBENAUEAAACAAAAAAAAAAAAAAAAA", ac: "" },
+    },
+    // Invalid case of only AC string- need core TC string
+    {
+      fidesString: ",1~2.3.4",
+      expected: { tc: "", ac: "" },
+    },
+    // Both TC and AC string
+    {
+      fidesString: "CPzvOIAPzvOIAGXABBENAUEAAACAAAAAAAAAAAAAAAAA.IAAA,1~2.3.4",
+      expected: {
+        tc: "CPzvOIAPzvOIAGXABBENAUEAAACAAAAAAAAAAAAAAAAA.IAAA",
+        ac: "1~2.3.4",
+      },
+    },
+    // With extra unexpected stuff
+    {
+      fidesString:
+        "CPzvOIAPzvOIAGXABBENAUEAAACAAAAAAAAAAAAAAAAA.IAAA,1~2.3.4,extrastuff",
+      expected: {
+        tc: "",
+        ac: "",
+      },
```

Comment:
is this actually what we expect? or would we rather still be able to parse `tc` and `ac` out? @NevilleS

Reply:
good question, my PR on the backend fails if handed a string that looks like this.

Reply:
Hmm... I'm OK with erroring in that case, although I think I prefer the idea of being (slightly) forwards-compatible here and just ignore the "extrastuff" suffix but still respect the TC & AC segments.

Reply:
To follow up from slack Allison, I'll be updating the backend to parse out TC and AC string here successfully and not erroring.

Reply:
thanks for the input—I also updated this to ignore `extrastuff` and successfully parse TC and AC out 👍